workers/unity/Assets/Playground/Scripts/Player/ConnectButtonClickSystem.cs

```diff
@@ -0,0 +1,93 @@
+using Improbable.Gdk.Core;
+using Unity.Collections;
+using Unity.Entities;
+using UnityEngine;
+using UnityEngine.UI;
+#if UNITY_ANDROID
+using Improbable.Gdk.Android;
+
+#endif
+
+namespace Playground
+{
+    [UpdateBefore(typeof(ConnectClientSystem))]
+    internal class ConnectButtonClickSystem : ComponentSystem
+    {
+        private struct Data
+        {
+            public readonly int Length;
+            [ReadOnly] public EntityArray Entity;
+            [ReadOnly] public ComponentDataArray<WorkerEntityTag> DenotesWorker;
+        }
+
+        private InputField connectParamInputField;
+        private Button connectButton;
+        private bool clicked;
+        private WorkerBase worker;
+        private Text errorField;
+
+        [Inject] private Data data;
+
+        protected override void OnCreateManager(int capacity)
+        {
+            base.OnCreateManager(capacity);
+            worker = WorkerRegistry.GetWorkerForWorld(World);
+            connectParamInputField = GameObject.Find("ConnectParam").GetComponent<InputField>();
+            connectButton = GameObject.Find("ConnectButton").GetComponent<Button>();
+            connectButton.onClick.AddListener(IsClicked);
+            errorField = GameObject.Find("ConnectionError").GetComponent<Text>();
+            if (!Application.isMobilePlatform)
+            {
+                connectParamInputField.gameObject.SetActive(false);
+            }
+        }
+
+        protected override void OnUpdate()
+        {
+            if (clicked)
+            {
+                for (var i = 0; i < data.Length; i++)
+                {
+                    PostUpdateCommands.AddComponent(data.Entity[i], new ConnectButtonClicked());
+                    connectButton.gameObject.SetActive(false);
+                    errorField.text = "";
+                    clicked = false;
+                }
+            }
+        }
+
+        private void IsClicked()
+        {
+            clicked = true;
+#if UNITY_ANDROID
+            if (Application.isMobilePlatform)
+            {
+                if (DeviceInfo.IsAndroidStudioEmulator() && connectParamInputField.text.Equals(""))
+                {
+                    worker.ConnectionConfig = ReceptionistConfig.CreateConnectionConfigForAndroidEmulator();
+                }
+                else
+                {
+                    SetConnectionParameters(connectParamInputField.text);
+                }
+            }
+#endif
+        }
+
+
+        private void SetConnectionParameters(string param)
+        {
+            if (IsIpAddress(param))
+            {
+                worker.ConnectionConfig = ReceptionistConfig.CreateConnectionConfigForPhysicalAndroid(param);
+            }
```

Comment:
we should show some kind of error, if it's not an ip address

Reply:
Resolved.

Reply:
Resolved in 17bf200